src/pextlib1.0/tracelib.c

```diff
@@ -584,6 +585,104 @@ static void sandbox_violation(int sock UNUSED, const char *path, sandbox_violati
     Tcl_UnsetVar(interp, "_sandbox_viol_path", 0);
 }
 
+static int fs_case_sensitive(int sock, const char *path) {
+    int ret = -1;
+
+    char *lowercase_path = strdup(path);
+    char *uppercase_path = strdup(path);
+
+    if ((!lowercase_path) || (!uppercase_path)) {
+        free(lowercase_path);
+        free(uppercase_path);
+
+        ui_error(interp, "memory allocation failure when duplicating string '%s'", path);
+        answer(sock, "#");
+        return ret;
+    }
+
+    for (char *tmp_ptr_low = lowercase_path,
+              *tmp_ptr_up  = uppercase_path;
+         *tmp_ptr_low && *tmp_ptr_up; /* Since both are copies of the same string,
+                                         should be the same anyway. */
+         ++tmp_ptr_low, ++tmp_ptr_up) {
+        *tmp_ptr_low = tolower(*tmp_ptr_low);
+        *tmp_ptr_up  = toupper(*tmp_ptr_up);
+    }
+
+    struct stat path_stat = { 0 },
+           lowercase_path_stat = { 0 },
+           uppercase_path_stat = { 0 };
+
+    if (-1 == lstat(path, &path_stat)) {
+        free(lowercase_path);
+        free(uppercase_path);
+
+        ui_error(interp, "unable to get information on file %s", path);
+        answer(sock, "#");
```

Comment:
I think you should answer `?` for non-existant files. `#` will cause the client to `abort(3)`, which may have strange side effects.